src/openrct2/scripting/bindings/game/ScConsole.hpp

```diff
@@ -9,58 +9,67 @@
 
 #pragma once
 
-#ifdef ENABLE_SCRIPTING
+#ifdef ENABLE_SCRIPTING_REFACTOR
 
     #include "../../../interface/InteractiveConsole.h"
-    #include "../../Duktape.hpp"
     #include "../../ScriptEngine.h"
 
 namespace OpenRCT2::Scripting
 {
-    class ScConsole
+    class ScConsole;
+    extern ScConsole gScConsole;
+
+    class ScConsole final : public ScBase
     {
     private:
-        InteractiveConsole& _console;
-
-    public:
-        ScConsole(InteractiveConsole& console)
-            : _console(console)
+        static JSValue clear(JSContext* ctx, JSValue thisVal, int argc, JSValue* argv)
         {
+            if (const auto console = gScConsole.GetOpaque<InteractiveConsole*>(thisVal))
+                console->Clear();
+            return JS_UNDEFINED;
         }
 
-        void clear()
+        static JSValue log(JSContext* ctx, JSValue thisVal, int argc, JSValue* argv)
         {
-            _console.Clear();
+            if (const auto console = gScConsole.GetOpaque<InteractiveConsole*>(thisVal))
+            {
+                std::string line;
+                for (int i = 0; i < argc; i++)
+                {
+                    if (i != 0)
+                        line.push_back(' ');
+                    line += Stringify(ctx, argv[i]);
+                }
+                console->WriteLine(line);
+            }
+            return JS_UNDEFINED;
         }
 
-        duk_ret_t log(duk_context* ctx)
+        static JSValue executeLegacy(JSContext* ctx, JSValue thisVal, int argc, JSValue* argv)
         {
-            std::string line;
-            auto nargs = duk_get_top(ctx);
-            for (duk_idx_t i = 0; i < nargs; i++)
+            if (const auto console = gScConsole.GetOpaque<InteractiveConsole*>(thisVal))
             {
-                auto arg = DukValue::copy_from_stack(ctx, i);
-                auto argsz = Stringify(arg);
-                if (i != 0)
-                {
-                    line.push_back(' ');
-                }
-                line += argsz;
+                JS_UNPACK_STR(str, ctx, argv[0]);
+                console->Execute(str);
             }
-            _console.WriteLine(line);
-            return 0;
+            return JS_UNDEFINED;
         }
 
-        void executeLegacy(const std::string& command)
+        static constexpr JSCFunctionListEntry funcs[] = {
+            JS_CFUNC_DEF("clear", 0, clear),
+            JS_CFUNC_DEF("log", 0, log),
+            JS_CFUNC_DEF("executeLegacy", 0, executeLegacy),
```

Comment:
Shouldn't this one be 1 argument? (`executeLegacy(command: string): void;` in the TS file)